sources/Outputs/src/Par.cpp

```diff
@@ -157,7 +169,18 @@ Par::writeGenerator(const algo::GeneratorDefinition& def, const std::string& bas
   set->addParameter(helper::buildParameter("generator_QMinTableFile", dirname_diagram.generic_string()));
   set->addParameter(helper::buildParameter("generator_QMinTableName", hashIdStr + constants::diagramMinTableSuffix));
 
-  set->addReference(helper::buildReference("generator_PNom", "p_pu", "DOUBLE"));
+  switch (def_.activePowerCompensation) {
+  case dfl::inputs::Configuration::ActivePowerCompensation::P:
+    set->addReference(helper::buildReference("generator_PNom", "p_pu", "DOUBLE"));
+    break;
+  case dfl::inputs::Configuration::ActivePowerCompensation::TARGET_P:
+    set->addReference(helper::buildReference("generator_PNom", "targetP_pu", "DOUBLE"));
+    break;
+  case dfl::inputs::Configuration::ActivePowerCompensation::PMAX:
```

Comment:
Add a comment or do nothing case for default, to explain that ther is no default case by definition of the enum.